### ISSUE TITLE
DX11 crash when changing max frame latency

### ISSUE THREAD

**Issue by mctb32** [DiligentGraphics/DiligentEngine#277]
I have a crash in every Diligent Sample when runnning the D3D11 backend. The crash happens when trying to change the "max frame latency" option to any value. DX12 does not crash when changing the latency (Vulkan and OGL don't have this option). Release/64 crashes, Debug/64 catches an assertion:

![Zrzut ekranu 2023-12-28 164416](https://github.com/DiligentGraphics/DiligentEngine/assets/36546756/d359d348-cb13-48c1-a2b6-1ac6cc21f480)

Call stack and variables:

![Zrzut ekranu 2023-12-28 165118](https://github.com/DiligentGraphics/DiligentEngine/assets/36546756/26f3358c-0de5-4c08-8a67-edc9af12f664)

Environment:

- Diligent latests master branch
- Windows 11 22H2 (22621.2861) 
- Visual Studio 2022 17.8.3, CMake 3.27.9
- GeForce RTX 4080 (driver 546.33)

**Comment by TheMostDiligent** [comment]
Can you try this on Windows 10? There have been quite a few issues reported on Windows 11

**Comment by mctb32** [comment]
> Can you try this on Windows 10? There have been quite a few issues reported on Windows 11

Yes. Just tested and indeed it's running fine on Win 10. But the Win 10 PC I tested on also had a different GPU (GTX 1070Ti) with a game ready driver (the Win 11 PC is running a studio driver), so OS was not the only difference.

One more thing I noticed - when running on Windows 11, the V-Sync checkbox in both DX11 and DX12 seems to have no effect, the fps is permanently locked to my refresh rate. In Vulkan and OpenGL it works fine. And on Win 10 it works correctly in all 4 backends.

**Comment by TheMostDiligent** [comment]
> the Win 10 PC I tested on also had a different GPU (GTX 1070Ti) with a game ready driver

This is unlikely the driver. Nobody has reported this issue on Win10.

> the V-Sync checkbox in both DX11 and DX12 seems to have no effect, the fps is permanently locked to my refresh rate

Yes, somebody else has also reported this problem. There appears to be quite a few issues on Win11, they should be fixed eventually.

**Comment by mctb32** [comment]
Thanks. I have some more info about the crash. It happens in the SampleApp::Render() after the max frame latency change. Apparently there is some issue with the swapchain and now it returns a NULL RTV and DSV. The debug output log seems to contain some relevant info:

![Zrzut ekranu 2023-12-29 183544](https://github.com/DiligentGraphics/DiligentEngine/assets/36546756/30a91e20-dea9-4725-9bf0-6c1493f9d9b2)

**Comment by TheMostDiligent** [comment]
@mctb32 Can you test if this issue still reproduces?
I tested it on Windows 11 10.0.26100 with RTX4090 and everything seems to be working fine in both Debug and Release modes for both D3D11 and D3D12. Also works properly on Windows 10.
Plus VSync off is also fixed now

**Comment by mctb32** [comment]
Great. I will check in the evening when I get home.

**Comment by mctb32** [comment]
@TheMostDiligent Looks good - vsync off seems to work like you said, and the crash when changing frame latency is also gone. Thanks!

**Comment by TheMostDiligent** [comment]
Thanks for confirming
